src/ctapipe/io/hdf5monitoringsource.py

```diff
@@ -0,0 +1,521 @@
+      399.66666667]]
+    [[ 0.01000023  0.1800003  -0.09000015 ... -0.12999916  0.1800003
+       0.07999992]
+     [ 0.2800007  -0.27000046  0.11000061 ...  0.04000092 -0.19000053
+      -0.4699993 ]]
+    [[False False False ... False False False]
+     [False False False ... False False False]]
+    True
+
+    Attributes
+    ----------
+    input_url: str
+        Path to the input monitoring file.
+    file: tables.File
+        File object
+    pixel_statistics: dict
+        Dictionary to hold pixel statistics tables
+    camera_coefficients: dict
+        Dictionary to hold camera coefficients
+    telescope_pointings: dict
+        Dictionary to hold telescope pointing information
+
+    """
+
+    input_url = Path(
+        help="Path to the HDF5 input file containing monitoring data."
+    ).tag(config=True)
+
+    overwrite_telescope_pointings = Bool(
+        False,
+        help=(
+            "Flag to overwrite the telescope pointing information from the monitoring data. "
+            "Different EventSource implementations may already provide this information."
+        ),
+    ).tag(config=True)
+
+    enforce_subarray_description = Bool(
+        True,
+        help=(
+            "Force the reading of the subarray description "
+            "from the input file if the subarray is not provided."
+        ),
+    ).tag(config=True)
+
+    def __init__(
+        self, subarray=None, input_url=None, config=None, parent=None, **kwargs
+    ):
+        """
+        MonitoringSource for monitoring files in the standard HDF5 data format
+
+        Parameters:
+        -----------
+        subarray : SubarrayDescription or None
+            Description of the subarray to use. If not provided, the subarray
+            will be read from the input file. The 'enforce_subarray_description' flag
+            determines whether a subarray description is required.
+        input_url : str
+            Path of the file to load
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        parent:
+            Parent from which the config is used. Mutually exclusive with config
+        kwargs
+        """
+
+        super().__init__(
+            subarray=subarray,
+            input_url=input_url,
+            config=config,
+            parent=parent,
+            **kwargs,
+        )
+        from ..monitoring import (
+            FlatfieldImageInterpolator,
+            FlatfieldPeakTimeInterpolator,
+            PedestalImageInterpolator,
+            PointingInterpolator,
+        )
+
+        # Check if the reading from the input file is enforced
+        if self.enforce_subarray_description:
+            # Read the subarray description from the file if required
+            self._subarray_from_file = SubarrayDescription.read(self.input_url)
```

Comment:
Since you here know that it is an HDF5 file, use `from_hdf`. This is much more efficient as it doesn't go through the `EventSource` machinery to get the subarray but directly reads it.

Reply:
I see! Thanks!